utils/githooks/pre-commit

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+# A git hook to validate  and correct the copyright date in source files.
+
+regex='(^\s*\*.*)((Copyright\s*)([0-9]{4})(-([0-9]{4}))?)(\s*(Intel.*$))'
+year=`date +%Y`
+errors=0
+
+exec 1>&2
+
+for file in `git diff --diff-filter=AM --name-only --cached -- '*.c' '*.h' 'Makefile'`
+do
+	read -r y1 y2 <<< `sed -nre "s/^.*$regex.*$/\4 \6/p" "$file"`
```

Comment:
(lint) In POSIX sh, here-strings are not supported. [SC2039]
(lint) Quote this to prevent word splitting. [SC2046]
(lint) Use $(..) instead of legacy `..`. [SC2006]